src/Router/BaseRouter.php

```diff
@@ -118,6 +118,8 @@ public function handle(ServerRequestInterface $request) : ResponseInterface
             $baseurl = $uri->withPath($baseurl)->withQuery("");
             $request = $request->withAttribute("baseurl", $baseurl->__toString());
 
+            // Reset factory object so that the base URL can be reset
+            $factory->reset();
```

Comment:
@AlexandraLivadas What's the value of the baseURL before the reset ??

Reply:
The value is an empty string

Reply:
@AlexandraLivadas `reset()` will clear important values along with the settings singleton that we need cleared at this point. this is too risky.

What I think would be better is to reset the settings variable directly in NDB_Factory::setBaseURL() function. so directly in that function, set the settings to null and next time it is instanciated it will have the correct value.

Please try that change and test it and if all is good I'll pass review